apps/mobile-wallet/src/App.tsx

```diff
@@ -203,11 +181,18 @@ const Main = ({ children, ...props }: ViewProps) => {
     network.status !== 'online' || verifiedFungibleTokenSymbols.withPriceHistory.length === 0
   )
 
-  const refreshAddressDataWhenPendingTxsConfirm = useCallback(() => {
-    dispatch(syncAddressesDataWhenPendingTxsConfirm({ addresses: addressesWithPendingTxs, pendingTxs }))
-  }, [addressesWithPendingTxs, dispatch, pendingTxs])
+  const checkForNewTransactions = useCallback(() => {
+    dispatch(syncLatestTransactions())
+  }, [dispatch])
+
+  const dataResyncNeeded =
+    nbOfAddresses > 0 && network.status === 'online' && !isLoadingLatestTxs && addressesStatus === 'uninitialized'
+
+  useEffect(() => {
+    if (dataResyncNeeded) checkForNewTransactions()
+  }, [addressesStatus, checkForNewTransactions, dataResyncNeeded])
 
-  useInterval(refreshAddressDataWhenPendingTxsConfirm, 5000, pendingTxs.length === 0)
+  useInterval(checkForNewTransactions, TRANSACTIONS_REFRESH_INTERVAL, !dataResyncNeeded)
 
```

Comment:
This whole "sync when pending" goes completely to the trash now that we have a simple, yet very efficient polling mechanism. Big performance boost from this one.